packages/filesystem/src/browser/file-tree/file-tree-widget.tsx

```diff
@@ -41,6 +42,17 @@ export class FileTreeWidget extends TreeWidget {
         this.toDispose.push(this.toCancelNodeExpansion);
     }
 
+    protected onActivateRequest(msg: Message): void {
+        super.onActivateRequest(msg);
+        const root = this.model.root;
+        if (CompositeTreeNode.is(root) && root.children.length === 1) {
```

Comment:
It should happen only if there is not a selection: If a user selected something before and collapsed first workspace root when he comes back we should not reopen it again.